src/components/ContactsPage.js

```diff
@@ -79,30 +91,88 @@ class ContactsPage extends React.Component {
     const updatedContacts = contacts.filter(x => !x.isChecked);
     this.setState({ contacts: updatedContacts });
   }
-  
-  numberOfSelectedContacts() {
+
+  numberOfSelectedContacts = () => {
     const { contacts } = this.state;
     return contacts.filter(x => x.isChecked).length || null;
   }
 
-  changeIsChecked(id) {
+  changeIsChecked = (id) => {
     const { contacts } = this.state;
     const contact = contacts.find(item => item.id === id);
     contact.isChecked = !contact.isChecked;
     this.setState({ contacts });
   }
 
-  updateLikes(id, step) {
+  updateLikes = (id, step) => {
     const { contacts } = this.state;
     contacts.find(item => item.id === id).likes += step;
     this.setState({ contacts });
   }
 
-  deleteContact(id) {
+  deleteContact = (id) => {
     const { contacts } = this.state;
     const updatedContacts = contacts.filter(c => c.id !== id);
     this.setState({ contacts: updatedContacts });
   }
+
+  deleteContactPopUp = (id) => {
+    this.setState({
+      popoup: {
+        showPopup: true,
+        message: WARNING_MESSAGES.DELETE_ONE,
+        approveFlag: {
+          flag: APPROVE_FLAGS.DELETE_ONE,
+          contactId: id
+        }
+      }
+    });
+  }
+
+  deleteSelectedContactsPopUp = () => {
+    this.setState({
+      popoup: {
+        showPopup: true,
+        message: WARNING_MESSAGES.DELETE_SELECTED,
+        approveFlag: {
+          flag: APPROVE_FLAGS.DELETE_SELECTED,
```

Comment:
If you move the 'contactId' one level above, you won't need to have the 'approveFlag' as an object. Directly add the flag value to it.

Reminder: Check the comment at the end of the merge request where I suggest renaming the 'approveFlag' property